lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -922,8 +922,43 @@ WitnessChecker::lookupValueWitnessesViaImplementsAttr(
   }
 }
 
-SmallVector<ValueDecl *, 4> 
-WitnessChecker::lookupValueWitnesses(ValueDecl *req, bool *ignoringNames) {
+bool WitnessChecker::isMemberOperator(FuncDecl *decl, Type type) {
+  auto *DC = decl->getDeclContext();
+  auto selfNominal = DC->getSelfNominalTypeDecl();
+
+  // Check the parameters for a reference to 'Self'.
+  bool isProtocol =
+      selfNominal && isa<ProtocolDecl>(selfNominal) &&
+      TC.conformsToProtocol(type, dyn_cast<ProtocolDecl>(selfNominal), DC,
+                            ConformanceCheckFlags::InExpression);
+
+  for (auto param : *decl->getParameters()) {
```

Comment:
You’re checking if at least one parameter is of the nominal type — but is that the right check?